near-sdk-sim/README.md

```diff
@@ -26,79 +34,430 @@ near-sdk = { git = "https://github.com/near/near-sdk-rs.git", tag="3.0.0" }
+
+// or rename it maybe
+use token::TokenContract as OtherNamedContract;
+```
+
+Now you can simplify the `init` & test code from the previous section:
+
 ```rust
+// in utils.rs
+use near_sdk_sim::{deploy, init_simulator, to_yocto, STORAGE_AMOUNT};
+use token::TokenContract;
+
+const CONTRACT_ID: &str = "contract";
+
+pub fn init() -> (UserAccount, ContractAccount<TokenContract>, UserAccount) {
+    let root = init_simulator(None);
+
+    let contract = deploy!(
+        contract: TokenContract,
+        contract_id: CONTRACT_ID,
+        bytes: &CONTRACT_WASM_BYTES,
+        signer_account: root,
+    );
+
+    let alice = root.create_user(
+        "alice".to_string(),
+        to_yocto("100") // initial balance
+    );
+
+    (root, contract, alice)
+}
+
+// in first_tests.rs
+use near_sdk_sim::{call, view};
+use crate::utils::init;
+
+#[test]
+fn simulate_some_view_function() {
+    let (root, contract, _alice) = init();
+
+    let actual: String = view!(
+        contract.view_something("some_value".to_string()),
+    ).unwrap_json();
+
+    assert_eq!("expected", actual);
+}
+
 #[test]
-fn simple_test() {
-      let master_account = init_simulator(None);
-      let initial_balance = to_yocto("100000");
-          // uses default values for deposit and gas
-      let contract_user = deploy!(
-          // Contract Proxy
-          contract: FungibleTokenContract,
-          // Contract account id
-          contract_id: "contract",
-          // Bytes of contract
-          bytes: &TOKEN_WASM_BYTES,
-          // User deploying the contract,
-          signer_account: master_account,
-          // init method
-          init_method: new(master_account.account_id(), initial_balance.into())
-      );
-    let alice = master_account.create_user("alice".to_string(), to_yocto("100"));
-
-    let res = call!(
-        master_account,
-        contract.transfer(alice.account_id.clone(), transfer_amount.into()),
-        deposit = STORAGE_AMOUNT
+fn simulate_some_change_method() {
+    let (root, contract, _alice) = init();
+
+    // uses default gas amount
+    let result = call!(
+        root,
+        contract.change_something("some_value".to_string()),
+        deposit = 1,
     );
 
-    assert!(res.is_ok());
+    assert!(result.is_ok());
+}
+```
+
+# Common patterns
+
+## Profile gas costs
+
+For a chain of transactions kicked off by `call` or `call!`, you can check the `gas_burnt` and `tokens_burnt`, where `tokens_burnt` will equal `gas_burnt` multiplied by the `gas_price` set in the genesis config. You can also print out `profile_data` to see an in-depth gas-use breakdown.
+
+```rust
+let outcome = some_account.call(
+    "some_contract",
+    "method",
+    &json({
+        "some_param": "some_value",
+    }).to_string().into_bytes(),
+    DEFAULT_GAS,
+    0,
+);
+
+println!(
+    "profile_data: {:#?} \n\ntokens_burnt: {}Ⓝ",
+    outcome.profile_data(),
+    (outcome.tokens_burnt()) as f64 / f64::powi(10.0, 24)
```

Comment:
```suggestion
    (outcome.tokens_burnt()) as f64 / 1e24f64
```

Reply:
Thanks, `1e24` works!

Checking this revealed that `profile_data` is now broken. I created https://github.com/near/near-sdk-rs/pull/339 to track this issue.